airflow/providers/databricks/hooks/databricks_sql.py

```diff
@@ -163,38 +163,43 @@ def run(
         :param return_last: Whether to return result for only last statement or for all after split
         :return: return only result of the LAST SQL expression if handler was provided.
         """
-        self.scalar_return_last = isinstance(sql, str) and return_last
+        self.descriptions = []
         if isinstance(sql, str):
             if split_statements:
-                sql = self.split_sql_string(sql)
+                sql_list = [self.strip_sql_string(s) for s in self.split_sql_string(sql)]
```

Comment:
I  added strip_sql_string here to be consistent with the approach where no split_statements is set. It was inconsistent. Apparently it's ok for Databricks to run a single statement without ";" so we should be ok doing so.